modules/gdscript/gdscript_parser.cpp

```diff
@@ -1997,7 +1997,37 @@ GDScriptParser::MatchBranchNode *GDScriptParser::parse_match_branch() {
 		push_error(R"(No pattern found for "match" branch.)");
 	}
 
-	if (!consume(GDScriptTokenizer::Token::COLON, R"(Expected ":" after "match" patterns.)")) {
+	bool has_guard = false;
+	if (match(GDScriptTokenizer::Token::FORWARD_ARROW)) {
+		// Pattern guard.
+		// Create block for guard because it also needs to access the bound variables from patterns, and we don't want to add them to the outer scope.
+		branch->guard_body = alloc_node<SuiteNode>();
+		for (const KeyValue<StringName, IdentifierNode *> &E : branch->patterns[0]->binds) {
```

Comment:
This line causes the editor crash if the pattern is not completed.

```gdscript
match x:
    var -> if y == 0:
```

(After applying the patch on the current master, line 2005 became line 2018.)

```
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v4.2.dev.custom_build (3e153f026ab9734df61631a96b412d40008e7919)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x3c4b0) [0x7f949c03c4b0] (??:0)
[2] CowData<GDScriptParser::PatternNode*>::get(int) const (/home/danil/godot-src/./core/templates/cowdata.h:155 (discriminator 7))
[3] Vector<GDScriptParser::PatternNode*>::operator[](int) const (/home/danil/godot-src/./core/templates/vector.h:96)
[4] GDScriptParser::parse_match_branch() (/home/danil/godot-src/modules/gdscript/gdscript_parser.cpp:2018)
[5] GDScriptParser::parse_match() (/home/danil/godot-src/modules/gdscript/gdscript_parser.cpp:1957)
```

Reply:
That's a bad copy-paste of mine, nice catch